api/v2/checluster_types.go

```diff
@@ -72,6 +73,15 @@ type CheClusterDevEnvironments struct {
 	// Trusted certificate settings.
 	// +optional
 	TrustedCerts TrustedCerts `json:"trustedCerts,omitempty"`
+	// The default editor to workspace create with. It could be a plugin ID or a URI.
+	// The plugin ID must have `publisher/plugin/version`.
+	// The URI must start from `http`.
+	// +optional
+	DefaultEditor string `json:"defaultEditor,omitempty"`
+	// Default components applied to DevWorkspaces.
+	// These default components are meant to be used when a Devfile does not contain any components.
+	// +optional
+	DefaultComponents []devfile.Component `json:"defaultComponents,omitempty"`
```

Comment:
@amisevsk I have some doubts about types (using ```[]devfile.Component```).
Could you confirm that I am going to use the right one?

Reply:
I think your approach is the best here. The alternative would be something like `[]devfile.ContainerComponent`, but this wouldn't allow having e.g. default volumes and would require setting a `name` for each component somewhere else. With `[]devfile.Component`, it should be possible to just set `currentDevWorkspace.spec.template.components = DefaultComponents` for ease.